src/util/records.py

```diff
@@ -99,12 +99,43 @@ def remove_view(cr, xml_id=None, view_id=None, silent=False, key=None):
             if cr.rowcount:
                 xml_id = "%s.%s" % cr.fetchone()
 
-    # From given or determined xml_id, the views duplicated in a multi-website
-    # context are to be found and removed.
-    if xml_id != "?" and column_exists(cr, "ir_ui_view", "key"):
-        cr.execute("SELECT id FROM ir_ui_view WHERE key = %s AND id != %s", [xml_id, view_id])
-        for [v_id] in cr.fetchall():
-            remove_view(cr, view_id=v_id, silent=silent, key=xml_id)
+    if xml_id != "?":
+        # From given or determined xml_id, the views duplicated in a multi-website
+        # context are to be found and removed.
+        if column_exists(cr, "ir_ui_view", "key"):
+            cr.execute("SELECT id FROM ir_ui_view WHERE key = %s AND id != %s", [xml_id, view_id])
+            for [v_id] in cr.fetchall():
+                remove_view(cr, view_id=v_id, silent=silent, key=xml_id)
+
+        # Occurrences of xml_id in the t-call of views are to be found and removed.
+        arch_col = (
+            get_value_or_en_translation(cr, "ir_ui_view", "arch_db")
+            if column_exists(cr, "ir_ui_view", "arch_db")
+            else "arch"
+        )
+        cr.execute(
+            r"""
+                SELECT iv.id, imd.module, imd.name
+                  FROM ir_ui_view iv
+             LEFT JOIN ir_model_data imd
+                    ON iv.id = imd.res_id
+                   AND imd.model = 'ir.ui.view'
+                 WHERE {} ~ '\yt-call=(["'']){}\1'
+            """.format(arch_col, xml_id)
```

Comment:
```suggestion
            format_query(cr, """
                SELECT iv.id, imd.module, imd.name
                  FROM ir_ui_view iv
             LEFT JOIN ir_model_data imd
                    ON iv.id = imd.res_id
                   AND imd.model = 'ir.ui.view'
                 WHERE {} ~ %s""", arch_col
            ),
            [r"""\yt-call=(["']){}\1""".format(xml_id)]
```

Reply:
Hello @aj-fuentes and @KangOl 

Changes have been made. Could you please confirm them?